### ISSUE TITLE
Virtual Nodes - Pod stuck in Pending state when referencing status.podIP from downward API

### ISSUE THREAD

**Issue by jonstelly** [Azure/AKS#2427]
**What happened**:
Deploying a pod to an AKS Virtual Node, the pod is stuck in the Pending state if a container has:

```
env:
  - name: SERVING_POD_IP
    valueFrom:
      fieldRef:
        fieldPath: status.podIP
```

I see the following error in the aci linux connector pod:

```
time="2021-07-05T22:34:00Z" level=warning msg="requeuing \"default/acitest-wv7lc\" due to failed sync"
  error="failed to sync pod \"default/acitest-wv7lc\" in the provider: 
  unsupported fieldPath: status.podIP" key=default/acitest-wv7lc method=handleQueueItem node=virtual-node-aci-linux
  operatingSystem=Linux provider=azure watchedNamespace= workerId=1
```

**What you expected to happen**:
For the pod to start up.  Below is a simple repro yaml for a job that displays the behavior.  You can comment out the SERVING_POD_IP variable and it starts as expected.  But the real use-case for this is Knative launching on the Virtual Nodes and they require that variable be set.

**How to reproduce it (as minimally and precisely as possible)**:

run: `kubectl apply -f ./acitest.yaml`
```
apiVersion: batch/v1
kind: Job
metadata:
  name: acitest
spec:
  backoffLimit: 1
  template:
    spec:
      containers:
        - name: ubuntu
          image: ubuntu:latest
          command: ["uname", "-a"]
          resources:
            limits:
              cpu: 1
              memory: 32Mi
            requests:
              cpu: 1
              memory: 32Mi
          env:
            - name: SERVING_POD
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.name
            - name: SERVING_POD_IP
              valueFrom:
                fieldRef:
                  fieldPath: status.podIP
      restartPolicy: Never
      nodeSelector:
        beta.kubernetes.io/os: linux
        kubernetes.io/role: agent
        type: virtual-kubelet   
      tolerations:
      - key: virtual-kubelet.io/provider
        operator: Exists
      - key: node.kubernetes.io/not-ready
        operator: Exists
        effect: NoExecute
        tolerationSeconds: 300
      - key: node.kubernetes.io/unreachable
        operator: Exists
        effect: NoExecute
        tolerationSeconds: 300
      - key: node.kubernetes.io/memory-pressure
        operator: Exists
        effect: NoSchedule    

```

**Anything else we need to know?**:
Knative in front of Virtual Nodes seems like it would be a very popular/common model.  I see another couple users with the same issue [here](https://docs.microsoft.com/en-us/answers/questions/352186/pods-stuck-in-pending-state-on-virtual-nodes.html) and another issue in this repository that was auto-closed: #1139

**Environment**:
- Kubernetes version (use `kubectl version`): 1.18.4 - 1.20.7
- Size of cluster (how many worker nodes are in the cluster?): 3 node, Virtual Nodes enabled
- General description of workloads in the cluster: Mixed dependencies + dotnet core components

**Comment by dprotaso** [comment]
@Azure/aks-leads @Azure/aks-pm any updates?

**Comment by palma21** [comment]
@peterpogorski

**Comment by dprotaso** [comment]
Hey folks any updates on whether this is something that can be fixed?

**Comment by dprotaso** [comment]
bump @peterpogorski @palma21

**Comment by dprotaso** [comment]
>  I will file a feature request on our engineering team to support this scenario.

Thanks @peterpogorski that's great to hear! Any rough estimate on when the eng team will be able to prioritize this issue?

**Comment by dprotaso** [comment]
Hey just following up regarding the priority of this issue? 

thanks

**Comment by ryanzhang-oss** [comment]
> Hey just following up regarding the priority of this issue?
> 
> thanks

Our current Azure virtual kubelet provider does not support this "valueFrom" semantics yet.  We will consider this in our future roadmap. Please stay tuned!

**Comment by gabemeola** [comment]
> > Hey just following up regarding the priority of this issue?
> > thanks
> 
> Our current Azure virtual kubelet provider does not support this "valueFrom" semantics yet. We will consider this in our future roadmap. Please stay tuned!

Hey Ryan, has this roadmap been released yet?
Pending on this ticket for https://github.com/knative/serving/pull/11636 closure

**Comment by dprotaso** [comment]
@ryanzhang-oss any update?

**Comment by khauser** [comment]
Good morning in the morning,
If you are not able to implement this, then please close this ticket or give some information about the status/progess. Keeping it open will let others depend on it.
